kpi/views.py

```diff
@@ -848,37 +1042,44 @@ def deployment(self, request, uid):
                 # coerced to a dict
                 return Response(dict(serializer.data))
         elif request.method == 'POST':
-            if asset.has_deployment:
-                raise exceptions.MethodNotAllowed(
-                    method=request.method,
-                    detail='Use PATCH to update an existing deployment'
-                    )
-            serializer = DeploymentSerializer(
-                data=request.data,
-                context={'asset': asset}
-            )
-            serializer.is_valid(raise_exception=True)
-            serializer.save()
-            # TODO: Understand why this 404s when `serializer.data` is not
-            # coerced to a dict
-            return Response(dict(serializer.data))
+            if not asset.can_be_deployed:
+                raise BadAssetTypeException(u"A form template can't be deployed")
```

Comment:
(sorry, same thing over and over) This message should probably match https://github.com/kobotoolbox/kpi/pull/1806/files#diff-43986855ae1015d5975b107cf1ee9869R691